### ISSUE TITLE
Error on Line 39 by executeing the script

### ISSUE THREAD

**Issue by p4rz1** [x4x/pywws2mysql#2]
Hello,

when I executeing the script I get the following error:

`Traceback (most recent call last):
  File "/opt/pywws2mysql/wws2mysql.py", line 39, in <module>
    from mysql_interface import db_interface as db
`

If I delete line 39, the script gets data from the weather station, but there is of course no connection to the database.

> `{'engine': 'psycopg2', 'database': 'wwsdb', 'host': 'localhost', 'user': 'root', 'table': 'wwsdb', 'password': 'Password'}
> start wws connection
> synchronization of wws timing
> No handlers could be found for logger "pywws.weather_station"
> timing error
> start wws connection
> synchronization of wws timing
> {'abs_pressure': 934.7,
>  'delay': 5,
>  'hum_in': 47,
>  'hum_out': 57,
>  'idx': datetime.datetime(2016, 12, 3, 14, 26, 59),
>  'rain': 150.29999999999998,
>  'status': 0,
>  'temp_in': 20.400000000000002,
>  'temp_out': 2.4000000000000004,
>  'wind_ave': 4.4,
>  'wind_dir': 10,
>  'wind_gust': 7.5}
> init database
> db conection error!
> start wws connection
> `
> 

Can someone help my to fix the error?

**Comment by x4x** [comment]
The missing file is in the load_db_module Branch.

It seams I messed up some files in the Branches.
Just checkout to the  load_db_module  Branch and tray again.

**Comment by myckei** [comment]
Same problem here.
Can you please be more specific with solution?
I dont know what to do with "load_db_module Branch". 
Thank you for your help!

Mickey

Phase2:
I have already found missing file, but still no working install:
{'engine': 'mysql.connector', 'database': 'wwsdb', 'host': '127.0.0.1', 'user':      'root', 'table': 'wwsdb', 'password': 'password'}
start wws connection
synchronization of wws timing
No handlers could be found for logger "pywws.weather_station"
{'abs_pressure': 982.2,
 'delay': 5,
 'hum_in': 52,
 'hum_out': 88,
 'idx': datetime.datetime(2017, 1, 1, 18, 52, 3),
 'rain': 4.5,
 'status': 0,
 'temp_in': 23.200000000000003,
 'temp_out': -5.6000000000000005,
 'wind_ave': 0,
 'wind_dir': 2,
 'wind_gust': 0.30000000000000004}
init database
db conection error!
start wws connection
synchronization of wws timing
{'abs_pressure': 982.3000000000001,
 'delay': 5,
 'hum_in': 52,
 'hum_out': 88,
 'idx': datetime.datetime(2017, 1, 1, 18, 57, 3),
 'rain': 4.5,
 'status': 0,
 'temp_in': 23.200000000000003,
 'temp_out': -5.7,
 'wind_ave': 0,
 'wind_dir': 2,
 'wind_gust': 0.7000000000000001}
init database
Exception AttributeError: "'mysql_interface' object has no attribute '_mysql_int     erface__mysqldb_a'" in <bound method mysql_interface.__del__ of <mysql_interface     .db_interface.mysql_interface object at 0xb67440f0>> ignored
db conection error!

mywws2mysql.ini:
[syslog]
ident: "wws2mysql"
logoption: syslog.LOG_PID
facility: syslog.LOG_LOCAL0

[mysql_db]
; engine examples:  mysql.connector, psycopg2 
engine: mysql.connector
user: root
password: password
host: 127.0.0.1
database: wwsdb
table: wwsdb

[file_output]
output: True

I have tried engine set to psycopg2, but same error :-/

**Comment by x4x** [comment]
I meant you go to the other branch:
```git checkout load_db_module```
But apart from the missing file there shouldn’t be any differents.


I‘m not totally sure but have you installed a database engine?
```pip install mysql-connector```
or
```pip install psycopg2```
Depending on what database you use.


You also need to manually create a user on the database and the table.
The table creation files are *Create_psql.sql* or *Create_mysql.sql*.

**Comment by myckei** [comment]
Thank you for support.
Missing mysql-connector was the goal. I use MySQL database - database, and table were created, database user is correct.
After install, mysql-connector imort seems to work, but another message is shown at the and of import process:
Traceback (most recent call last):
  File "wws2mysql.py", line 278, in <module>
    sys.exit(main())
  File "wws2mysql.py", line 228, in main
    last_date = last_date - datetime.timedelta(minutes= block['delay'],seconds=             0)
TypeError: unsupported type for timedelta minutes component: NoneType


Thanks for your help.

**Comment by x4x** [comment]
This happens wen there is no last entry in the Database.
I thought i fixed this :(

as a workaround you can make a first default entry in the database:

`
INSERT INTO wwsdb (
abs_pressure
,delay
,hum_in
,hum_out
,idz
,ptr
,rain
,status
,temp_in
,temp_out
,wind_ave
,wind_dir
,wind_gust
)  VALUES (
934.7,
5,
47,
57, 
'2016-12-3 14:26:59',
0, 
150.29,
0,
20.4,
2.4,
4.4,
10,
7.5
);
`

**Comment by myckei** [comment]
Thank you fof answer.
I have changed two things:
1. Insert row in databse.
2. Ran command python /opt/pywws2mysql/wws2mysql.py not under root, cause I USB port is assigned to another user.

And now, import is working like a charm!
One thing, but minority, is, that imported rows are sometimes "strange" in imported time. 
Rows in database are in 5 min intervals, but when import starts, there are values for example:
10:05:00
10:10:00
10:10:01
10:15:01

So there is one second shift. It is not so important, but interesting :-)
